flutter/lib/state/app_state.dart

```diff
@@ -0,0 +1,233 @@
+import 'package:mlperfbench/state/task_runner.dart';
+import 'package:mlperfbench/store.dart';
+import '../benchmark/benchmark.dart';
+
+enum AppStateEnum {
+  downloading,
+  resourceError,
+  ready,
+  running,
+  aborting,
+}
+
+class AppStateHelper {
+  final Store store;
+  final TaskListManager taskListManager;
+  final ResourceManager resourceManager;
+  final ConfigManager configManager;
+  final TaskRunner taskRunner;
+  final LastResultManager lastResultManager;
+
+  AppStateHelper({
+    required this.store,
+    required this.taskListManager,
+    required this.resourceManager,
+    required this.configManager,
+    required this.taskRunner,
+    required this.lastResultManager,
+  });
+
+  Future<void> _loadResources() async {
+    // TODO refactor this method
+
+    final newAppVersion =
+        '${BuildInfoHelper.info.version}+${BuildInfoHelper.info.buildNumber}';
+    var needToPurgeCache = store.previousAppVersion != newAppVersion;
+    store.previousAppVersion = newAppVersion;
+
+    await Wakelock.enable();
+    print('start loading resources');
+    await resourceManager.handleResources(
+      taskListManager.taskList.listResources(
+        modes: [taskRunner.perfMode, taskRunner.accuracyMode],
+        skipInactive: false,
+      ),
+      needToPurgeCache,
+    );
+    print('finished loading resources');
+    await Wakelock.disable();
+  }
+
+  Future<void> _runTasks() async {
+    // we want last result to be null in case an exception is thrown
+    lastResultManager.value = null;
+
+    // disable screen sleep when benchmarks is running
+    await Wakelock.enable();
+
+    final startTime = DateTime.now();
+    final logDirName = startTime.toIso8601String().replaceAll(':', '-');
+    final currentLogDir = '${resourceManager.resourceDir}/logs/$logDirName';
+
+    try {
+      final res = await taskRunner.runBenchmarks(
+          taskListManager.taskList, currentLogDir);
+      if (res != null) {
+        lastResultManager.value = res;
+        await resourceManager.resultManager.addResult(res);
+      }
+      print('Benchmarks finished');
+    } finally {
+      if (currentLogDir.isNotEmpty && !store.keepLogs) {
+        await Directory(currentLogDir).delete(recursive: true);
+      }
+
+      await Wakelock.disable();
+    }
+  }
+}
+
+// All methods in AppState should have the following properties:
+//   - Public methods: return type `void`, public methods must not be awaited
+//   - Public methods: exceptions are never thrown
+//   - All methods: notifyListeners() call outside of `state` setter is forbidden
+//   - All methods: if notifyListeners() is needed when `state` field doesn't change,
+//        the class that contains the data that requires UI update should implement ChangeNotifier instead
+//   - All methods: complex logic should be moved to other classes, methods in AppState should only manage `state`
+//
+// Any methods or getters that don't satisfy these requirements should be subject to refactoring.
```

Comment:
@anhappdev, the #556 issue and this whole PR in particular was for these rules.
Other changes may be subject to discussions, but I think these few lines are very important.
Everything else in this PR was changed to make this possible. Even though there are still few little things left in this class that don't obey these rules, now it will be easy to fix it.